tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/message/TraineeMessageListenerTest.java

```diff
@@ -29,25 +31,40 @@ void setUp() {
   }
 
   @Test
-  void shouldSaveSignedCojWhenEventValid() {
+  void shouldSaveSignedCojWhenEventValidWithCmId() {
     ConditionsOfJoiningDto dto = new ConditionsOfJoiningDto();
     dto.setSignedAt(SIGNED_AT);
     dto.setVersion(GoldGuideVersion.GG9);
-    ConditionsOfJoiningSignedEvent event = new ConditionsOfJoiningSignedEvent(ID, dto);
+    ConditionsOfJoiningSignedEvent event = new ConditionsOfJoiningSignedEvent(
+        CURRICULUM_MEMBERSHIP_ID, dto);
 
     listener.receiveMessage(event);
 
-    verify(service).save(ID, dto);
+    verify(service).save(CURRICULUM_MEMBERSHIP_ID, dto);
+  }
+
+  @Test
+  void shouldSaveSignedCojWhenEventValidWithPmId() {
+    ConditionsOfJoiningDto dto = new ConditionsOfJoiningDto();
+    dto.setSignedAt(SIGNED_AT);
+    dto.setVersion(GoldGuideVersion.GG9);
+    ConditionsOfJoiningSignedEvent event = new ConditionsOfJoiningSignedEvent(
+        PROGRAMME_MEMBERSHIP_ID, dto);
```

Comment:
While I'm nit-picking, this is now a duplicate block.  Following the pattern elsewhere, it would be moved to a "create" method.

Reply:
Personally I lean towards duplication over `createXyz` methods for unit tests, unless we're also going to write a test for the createXyz method? :trollface: 

I find it easier to understand what the test is expecting when the initialisation is all there upfront. Otherwise I have to scroll around to see what the creation method is doing.